ext/standard/var.c

```diff
@@ -600,7 +610,24 @@ PHPAPI void php_var_export_ex(zval *struc, int level, smart_str *buf) /* {{{ */
 			if (myht) {
 				if (!is_enum) {
 					ZEND_HASH_FOREACH_KEY_VAL_IND(myht, index, key, val) {
+						/* data is IS_PTR for properties with hooks. */
+						zval tmp;
+						if (UNEXPECTED(Z_TYPE_P(val) == IS_PTR)) {
+							zend_property_info *prop_info = Z_PTR_P(val);
+							if ((prop_info->flags & ZEND_ACC_VIRTUAL) && !prop_info->hooks[ZEND_PROPERTY_HOOK_GET]) {
+								continue;
```

Comment:
If property doesn't have a ``set`` hook, then ``var_export()`` should not write its value.
It cannot be restored anyway.

Reply:
This is an interesting remark. In theory, `__set_state` could still do something with it, but it's indeed true that it might not bee needed. We're discussing this case.

Reply:
We decided not to change anything here now, given that it is specified this way. FWIW, `var_export()` is pretty broken for objects anyway. https://3v4l.org/9mrMp